hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -177,6 +179,16 @@ func (r *NodePoolReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 				return ctrl.Result{}, fmt.Errorf("failed to remove finalizer from nodepool: %w", err)
 			}
 		}
+
```

Comment:
The cached DV needs to be deleted in the `r.delete()` function that is called in this if statement. FOr external infra, we can't depend on the namespace cleanup or any sort of owner references to cleanup the DV on the external clusters for us. We have to do it in the controller

Reply:
Good catch!

Currently, we're using the hostedCluster's infraID field to name the cache DV, so we have the same problem as in the external client deletion. 

I think the best solution for that is to add the cache DV name to a new optional filed under the nodePoll **status**. Then we can easily delete it. In addition, when reading the DV with the external client, we can read it directly - not by a List, and then we also sole the informer issue.

WDYT?